File: service/locator/Locator/Locator/Location.swift
//
//  Location.swift
//  Locator
//
//  Created by Matthias Geisler on 23.02.24.
//

import Foundation

@objc public class Location: NSObject {
    public let longitude: Double
    public let latitude: Double
    
    public init(longitude: Double, latitude: Double) {
        self.longitude = longitude
        self.latitude = latitude
    }
}

File: data/weather/src/commonMain/kotlin/io/bitpogo/keather/data/weather/model/Location.kt
/*
 * Copyright (c) 2024 Matthias Geisler (bitPogo) / All rights reserved.
 *
 * Use of this source code is governed by Apache v2.0
 */

package io.bitpogo.keather.data.weather.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Location(
    val name: String,
    val region: String,
    val country: String,
    @SerialName("lat")
    val latitude: Double,
    @SerialName("lon")
    val longitude: Double,
    @SerialName("tz_id")
    val timezoneId: String,
    @SerialName("localtime")
    val localTime: String,
    @SerialName("localtime_epoch")
    val localTimestamp: Long,
)
